internal/resources/dataprotection/scope/scope_schema.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package scope
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"golang.org/x/exp/slices"
+
+	dataprotectionmodels "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/cluster/dataprotection"
+	dataprotectioncgmodels "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/clustergroup/dataprotection"
+)
+
+type (
+	Scope int64
+	// ScopedFullname is a struct for all types of policy full names.
+	ScopedFullname struct {
+		Scope                Scope
+		FullnameCluster      *dataprotectionmodels.VmwareTanzuManageV1alpha1ClusterDataprotectionFullName
+		FullnameClusterGroup *dataprotectioncgmodels.VmwareTanzuManageV1alpha1ClustergroupDataprotectionFullName
+	}
+)
+
+func ConstructScope(d *schema.ResourceData) (scopedFullnameData *ScopedFullname) {
```

Comment:
Since the converter is used do you really need these construct methods?

Reply:
Not strictly required but extracting the scope & fullname to a struct makes it easier to access the fields rather than using `schema.ResourceData` directly.